tools/build/scanCode.py

```diff
@@ -162,32 +300,65 @@ def is_not_symlink(path):
         return None
 
 
-def line_checks(checks):
-    """Turn file-based check into line-by-line checks on each file."""
-    def run_line_checks(file_path):
-        errors = []
-        ln = 0
-        with open(file_path) as fp:
-            for line in fp:
-                ln += 1
-                for check in checks:
-                    err = check(line)
-                    if err is not None:
-                        errors.append((ln, err))
-        return errors
-    return run_line_checks
+# Note: this function must appear after all "check" functions are defined
+def read_path_inclusions(config):
+    """Read the list of paths to include in scan tests."""
+    inclusion_dict = get_config_section_dict(config, SECTION_INCLUDE)
+    # vprint("inclusion_dict: " + str(inclusion_dict))
+
+    for key in inclusion_dict:
+        all_checks = inclusion_dict[key]
+        # strip off all whitespace, regardless of index
+        all_checks = all_checks.replace(' ', '')
+        # retrieve the names of all functions to scan for
+        # the respective filename (wildcards allowed)
+        function_names = all_checks.split(',')
+        file_check_fxs = []
+        line_check_fxs = []
+        for fname in function_names:
+            try:
+                fx = globals()[fname]
+                if fname in FILE_CHECK_FUNCTIONS:
+                    file_check_fxs.append(fx)
+                elif fname in LINE_CHECK_FUNCTIONS:
+                    line_check_fxs.append(fx)
+            except Exception:
```

Comment:
Depending on the data structure I was getting Exceptions here when adding functions when authoring the code.  As for generalized Exception handling,  it should be fine for check for non-granular exceptions types for graceful exit.  The code works, if you wish to fine tune each line, feel free to after getting the actual function in, but this is starting to feel "over the top".